src/gui/threejsScene.js

```diff
@@ -821,6 +839,38 @@ import { ViewFrustum, frustumVertexShader, frustumFragmentShader, MAX_VIEW_FRUST
         };
     };
 
+    // source: https://stackoverflow.com/questions/29758233/three-js-check-if-object-is-still-in-view-of-the-camera
+    exports.isObjectOnScreen = function(objectPosition) {
```

Comment:
It is probably worth noting that this checks if the object's origin is on-screen rather than the entire object. Might be better to rename it `isPointOnScreen`?